### ISSUE TITLE
TabNav cannot be properly assigned `roles`

### ISSUE THREAD

**Issue by mattcosta7** [primer/react#1622]
**Describe the bug**

When using a `<TabNav />`, no `tablist` and `tab` roles are assigned directly. 

Due to the implementation, attempting to provide this manually creates invalid aria role semantics, since the rest props of TabNav are applied to the wrapper and not the 'nav' element, while the link roles are applied directly

current jsx => Markup
```
<TabNav role="tablist"><TabNav.Link role="tab /></TabNav>
=>
<div role="tablist"><nav><a role="tab /></nav></div>
```

expected

```
<TabNav role="tablist"><TabNav.Link role="tab /></TabNav>
=>
<div><nav role="tablist"><a role="tab /></nav></div>
```

or that these are applied by default?

**Comment by mattcosta7** [comment]
keep

**Comment by owenniblock** [comment]
We probably want to shift the elements around a little here so it looks like this:

```
<TabNav role="tablist"><TabNav.Link role="tab /></TabNav>
=>
<nav><div role="tablist"><a role="tab /></div></nav>
```

Since [nav does not permit changing `role`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav) but we don't want to lose the ability to navigate to the `nav` element using a screen reader. There's a [memex issue](https://github.com/github/accessibility-audits/issues/2438#issuecomment-1082074916) where we discuss this reordering if you want additional context.

I'm worried about applying this role by default since [tab elements cannot contain interactive elements](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role) and we certainly use that pattern in memex.

There's an argument to be made that we should enforce this component to use `tablist` and `tab` and add restrictions to enforce only text within the `tab` element but that will require some discussion and we'd need to provide recommendations for an alternative component where users are trying to create functionality like the memex tabs.

**Comment by lesliecdubs** [comment]
👋🏻 @owenniblock hi! I see there's a new draft PR related to this issue. Are you actively working on this?

**Comment by owenniblock** [comment]
@lesliecdubs it's blocked waiting for a Primer React release: https://github.com/primer/react/pull/2131

**Comment by lesliecdubs** [comment]
Got it, we'll work on getting that out this week. I'm assigning this issue to you in the meantime, but if that's incorrect please feel free to unassign. Thanks for moving this forward!

**Comment by mattcosta7** [comment]
@lesliecdubs I believe this one is done!
